x/concentrated-liquidity/swaps_tick_cross_test.go

```diff
@@ -0,0 +1,741 @@
+}
+
+// assertPositionOutOfRange a helper to assert that a position with the given lowerTick and upperTick is out of range.
+func (s *SwapTickCrossTestSuite) assertPositionOutOfRange(poolId uint64, lowerTick int64, upperTick int64) {
+	pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	isInRange := pool.IsCurrentTickInRange(lowerTick, upperTick)
+	s.Require().False(isInRange, "currentTick: %d, lowerTick %d, upperTick: %d", pool.GetCurrentTick(), lowerTick, upperTick)
+}
+
+// assertPositionRangeConditional a helper to assert that a position with the given lowerTick and upperTick is in or out of range
+// depending on the isOutOfRangeExpected flag.
+func (s *SwapTickCrossTestSuite) assertPositionRangeConditional(poolId uint64, isOutOfRangeExpected bool, lowerTick int64, upperTick int64) {
+	if isOutOfRangeExpected {
+		s.assertPositionOutOfRange(poolId, lowerTick, upperTick)
+	} else {
+		s.asserPositionInRange(poolId, lowerTick, upperTick)
+	}
+}
+
+// swapZeroForOneLeft swaps amount in the left (zfo) direction of the swap.
+// Asserts that no error is returned.
+func (s *SwapTickCrossTestSuite) swapZeroForOneLeft(poolId uint64, amount sdk.Coin) {
+	pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	s.FundAcc(s.TestAccs[0], sdk.NewCoins(amount))
+	_, err = s.App.ConcentratedLiquidityKeeper.SwapExactAmountIn(s.Ctx, s.TestAccs[0], pool, amount, pool.GetToken1(), sdk.ZeroInt(), sdk.ZeroDec())
+	s.Require().NoError(err)
+}
+
+// swapOneForZeroRight swaps amount in the right (ofz) direction of the swap.
+// Asserts that no error is returned.
+func (s *SwapTickCrossTestSuite) swapOneForZeroRight(poolId uint64, amount sdk.Coin) {
+	pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	s.FundAcc(s.TestAccs[0], sdk.NewCoins(amount))
+	_, err = s.App.ConcentratedLiquidityKeeper.SwapExactAmountIn(s.Ctx, s.TestAccs[0], pool, amount, pool.GetToken0(), sdk.ZeroInt(), sdk.ZeroDec())
+	s.Require().NoError(err)
+}
+
+// asserPoolLiquidityEquals a helper to assert that the liquidity of a pool is equal to the expected value.
+func (s *SwapTickCrossTestSuite) asserPoolLiquidityEquals(poolId uint64, expectedLiquidity sdk.Dec) {
+	pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	s.Require().Equal(expectedLiquidity, pool.GetLiquidity())
+}
+
+// asserPoolTickEquals a helper to assert that the current tick of a pool is equal to the expected value.
+func (s *SwapTickCrossTestSuite) asserPoolTickEquals(poolId uint64, expectedTick int64) {
+	pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	s.Require().Equal(expectedTick, pool.GetCurrentTick())
+}
+
+// TestSwapOutGivenIn_Tick_Initialization_And_Crossing tests that ticks are initialized and updated correctly
```

Comment:
Added moving large comments like this one to the `testing_strategy.md` task list here: #5570